README.md

```diff
@@ -75,6 +75,30 @@ This function can also be used in hiera files, for example to set class paramete
 some_class::password: "%{lookup('important-secret')}"
 ```
 
+You can use a fact to specify different vaults for different groups of nodes, for example:
+
+```yaml
+- name: 'Azure Key Vault Secrets per environment'
+    lookup_key: azure_key_vault::lookup
+    options:
+      vault_name: "%{facts.environment}-vault"
+      vault_api_version: '2016-10-01'
+      metadata_api_version: '2018-02-01'
+```
+
+For extra security consider including the name of the vault as an
+[extension in the certificate request](https://puppet.com/docs/puppet/latest/ssl_attributes_extensions.html)
+so it cannot be changed:
+
+```yaml
+- name: 'Azure Key Vault Secrets from trusted fact'
```

Comment:
Either get rid of this example OR leave it saying this is another way you could ensure nodes use the appropriate vault for secrets.